internal/cmd/login.go

```diff
@@ -146,29 +146,44 @@ func login(options loginCmdOptions) error {
 				return err
 			}
 		}
-
 	}
 
 	return loginToInfra(client, loginReq)
 }
 
 func loginToInfra(client *api.Client, loginReq *api.LoginRequest) error {
 	loginRes, err := client.Login(loginReq)
-	if api.ErrorStatusCode(err) == http.StatusUnauthorized {
```

Comment:
Just checking `StatusUnauthorized` isn't sufficient as there may be other reasons why login will fail. One example is OIDC login errors which never actually returns unauthorized. Instead, it may return 404 Not Found if the provider was removed between starting and completing login.

Reply:
We should be normalizing all of those errors on the server to always return unauthorized.

By removing this check we will be misleading users. Ex: if the error is actually a connection error we should not say "your access key may be invalid". That's not going to help them debug the problem.

I think this needs to be `if api.ErrorStatusCode(err) == http.StatusUnauthorized`, and we should audit the `Login` endpoint to make sure it's returning proper status codes in all cases.

If we need to, we can add other `if api.ErrorStatusCode(err) == http.StatusNotFound`, or we could add other fields to the response error to provide more fine grained information about the error.

Reply:
You're probably right. I took this approach since it was the path of least resistance. In hindsight, there are other scenarios I didn't consider. Will revert to check status code explicitly with additional checks for the cases I mentioned